examples/stable_diffusion_v2/cldm/cldm.py

```diff
@@ -398,3 +422,31 @@ def configure_optimizers(self):
             params += list(self.model.diffusion_model.out.get_parameters())
         opt = nn.optim.adam.AdamWeightDecay(params, learning_rate=lr)
         return opt
+
+    def p_losses(self, x_start, c_concat, c_crossattn, t, noise=None):
+        noise = ms.numpy.randn(x_start.shape)
+        x_noisy = self.q_sample(x_start=x_start, t=t, noise=noise)
+        model_output = self.apply_model(x_noisy, t, c_concat, c_crossattn)
+
+        if self.parameterization == "x0":
+            target = x_start
+        elif self.parameterization == "eps":
+            target = noise
+        elif self.parameterization == "velocity":
+            # target = sqrt_alpha_cum * noise - sqrt_one_minus_alpha_prod * x_start
+            target = self.get_velocity(x_start, noise, t)  # TODO: parse train step from randint
+        else:
+            raise NotImplementedError()
+
+        loss_simple = self.get_loss(model_output, target, mean=False).mean([1, 2, 3])
+
+        logvar_t = self.logvar[t]
+        loss = loss_simple / ops.exp(logvar_t) + logvar_t
+        loss = self.l_simple_weight * loss.mean()
+
+        # NOTE: original_elbo_weight is never set larger than 0. Diffuser remove it too. Let's remove it to save mem.
+        # loss_vlb = self.get_loss(model_output, target, mean=False).mean((1, 2, 3))
+        # loss_vlb = (self.lvlb_weights[t] * loss_vlb).mean()
+        # loss += (self.original_elbo_weight * loss_vlb)
```

Comment:
remove annotations

Reply:
This part is from the original file. Just keep it in case further needed.